packages/backend/src/open_payments/wallet_address/key/service.test.ts

```diff
@@ -129,11 +146,48 @@ describe('Wallet Address Key Service', (): void => {
       }
 
       const key = await walletAddressKeyService.create(keyOption)
+      assert.ok(!isWalletAddressKeyError(key))
 
       const revokedKey = await walletAddressKeyService.revoke(key.id)
       await expect(walletAddressKeyService.revoke(key.id)).resolves.toEqual(
         revokedKey
       )
     })
   })
+
+  describe('Unrevoke Wallet Address Keys', (): void => {
+    test('Creates a new key unrevoked', async (): Promise<void> => {
+      const keyOption = {
+        walletAddressId: walletAddress.id,
+        jwk: TEST_KEY
+      }
+
+      const key = await walletAddressKeyService.create(keyOption)
+      assert.ok(!isWalletAddressKeyError(key))
+      await walletAddressKeyService.revoke(key.id)
+
+      const unrevokedKey = await walletAddressKeyService.create(keyOption)
+
+      assert.ok(!isWalletAddressKeyError(key))
+
+      expect(unrevokedKey).toMatchObject({
+        jwk: key.jwk,
+        walletAddressId: key.walletAddressId,
+        revoked: false
+      })
+    })
+
+    test('Returns DuplicateKey when the key already exists and it is unrevoked', async (): Promise<void> => {
```

Comment:
Looks like you are already handling this in the "'cannot add duplicate key to a wallet address'" test.